### ISSUE TITLE
Feature Request: Ability to set relative python file paths in the reports

### ISSUE THREAD

**Issue by matroscoe** [astral-sh/ruff#7551]
This is happening with the latest version of ruff 
```shell
$ ruff --version 
> ruff 0.0.290
```

Trying to integrate ruff into sonar and it is having a hard time parsing the json output from the following command: 
```shell
ruff --exit-zero --output-file "ruff_report.json" --format json ./<module> ./tests
```
The issue seems to be that the ruff report when run in a GitHub action includes the full path so something like: 
```
/home/runner/work/<repo>/<namespace>/<module>/<module>/<module>/<python-file>.py
```
if would be nice to have a `relative` flag so that the report can just use the `<repo>/<namespace>/<module>/<module>/<module>/<python-file>.py` portion. This is the same behavior as the [relative flag in coverage.py](https://coverage.readthedocs.io/en/latest/config.html#run-relative-files)

**Comment by matroscoe** [comment]
for more background we are trying to have the report generated ingested into `sonarcloud` as a backup to pre-commit not working or being missed when it is run and sonar seems to be having problems with the fully justified paths that are generated when it is run as an action: https://community.sonarsource.com/t/importing-mypy-report-into-sonarcloud-from-pre-commit/101495/8

**Comment by zanieb** [comment]
I think that the JSON output just uses absolute paths (whereas our default output uses relative paths). Would the addition of a relative path field in our JSON output suffice?

**Comment by charliermarsh** [comment]
Hmm, how does ESLint solve this problem? They use absolute paths too.

**Comment by matroscoe** [comment]
I think it would, the absolute paths aren't an issue for us locally it is only causing problems when they are being run as a part of a GitHub action.

**Comment by philipp-horstenkamp** [comment]
SonarQube can now read ruff reports. If Sonar is only provided an artifact it needs relativ paths.
So relativ paths or the would be great!

https://community.sonarsource.com/t/support-ruff-linter-report-files/79714/9

**Comment by philipp-horstenkamp** [comment]
I used the ruff pylint style export and imported them as pylint issues. This is not a thing of beauty. But it works.
But yes i agree. A json export with relativ paths would be nice.

**Comment by crimsonknave** [comment]
I'm currently using the following workaround for an action where we run ruff inside a docker container and the paths don't match. My app is installed in `/app/` in the dockerfile. The sed strips that out to give a relative path. I believe that you should be able to do something similar for the json file, but `jq` might be a better tool there.

```bash
ruff check --output-format=github --output-file=ruff.txt
cat ruff.txt | sed 's/\/app\///'
```

**Comment by PhorstenkampFuzzy** [comment]
We are executing ruff as a fallback in jenkins and import the reports into SonarQube for ease of visualisation / tracking.
The current implementation with exporting in the pylint format works but isn't quiet as smooth as it could be.
Is there any way this could be implemented soonish? 🙇

**Comment by badrobit** [comment]
hey, was checking in to see if there had been any movement on this? it is mainly an issue when feeding into tools for overall quality monitoring such as `SonarCloud`

**Comment by OliverKleinBST** [comment]
Still a problem for me too
